cmsl1t/analyzers/BaseAnalyzer.py

```diff
@@ -11,18 +11,20 @@ class BaseAnalyzer(object):
     A Base class to be used by the various analyzers
     """
 
-    def __init__(self, name, config, **kwargs):
-        self.name = name
-        self.output_folder = config.get('output', 'folder')
-        self.plots_folder = config.get('output', 'plots_folder')
-        self.config = config
+    def __init__(self, **kwargs):
+        self.name = kwargs.pop('name')
+        self.output_folder = kwargs.pop('output_folder')
+        self.plots_folder = kwargs.pop('plots_folder')
+        self.all_plots = []
+        self.puBins = kwargs.pop('puBins')
+        self.thresholds = kwargs.pop('thresholds')
```

Comment:
I can imagine some analyzers not wanting to use the `puBins` or `thresholds` but by using `pop` here with no default, you'll get an exception in those cases.  Since you capture `kwargs` as `self.params`, might it make sense to just drop these two lines and let people get them from the params dict anyway?

Reply:
Yes to put in some defaults with `pop`.

Getting `puBins` from params would mean that we have to move these from a global scope to an analyzer scope. If that is OK, I can implement it.

Reply:
But that must already be the case, no?  `self.params` is just assigned to `kwargs` two lines below.  And on this line you get `thresholds` by popping it from the `kwargs` dict.  In other words, if you don't pop `pubins` and `thresholds` on lines 19 and 20, they'd appear in `self.params`.

Reply:
OK, done